File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int lastoccurence(vector<int>&nums,int target){
         int res =-1;
        int l =0; int h = nums.size()-1;
        while(l<=h){
            int mid = l+(h-l)/2;
            if(nums[mid]==target){
                res=mid;
                l=mid+1;
            }
            else if(nums[mid]>target){
                h = mid-1;
            }
            else{
                l = mid+1;
            }
        }
        return res;
    }
    int firstoccurence(vector<int>&nums,int target){
         int res =-1;
        int l =0; int h = nums.size()-1;
        while(l<=h){
            int mid = l+(h-l)/2;
            if(nums[mid]==target){
                res=mid;
                h=mid-1;
            }
            else if(nums[mid]>target){
                h = mid-1;
            }
            else{
                l = mid+1;
            }
        }
        return res;
    }
    vector<int> searchRange(vector<int>& nums, int target) {
       int firsocc= firstoccurence(nums,target);
       int lastocc = lastoccurence(nums,target);
       return {firsocc,lastocc};
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def firstOccurence(self, nums:List[int],target:int)-> int:
        left =0
        right =len(nums)-1
        index =-1
        while(left<=right):
            mid = left + (right - left) // 2
            if(nums[mid]==target):
                index = mid
            if(nums[mid]>=target):
                right = mid-1
            else:
                left = mid+1
        return index     

    def lastOccurence(self, nums:List[int],target:int)-> int:
        left =0
        right =len(nums)-1
        index = -1
        while(left<=right):
            mid = left + (right - left) // 2
            if(nums[mid]==target):
                index = mid
            if(nums[mid]<=target):
                left = mid+1
            else:
                right = mid-1
        return index
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        first = self.firstOccurence(nums,target)
        last = self.lastOccurence(nums,target)
        return [first,last]
        